src/rsa.rs

```diff
@@ -151,7 +98,43 @@ rsa_pkcs1_padding!(RSA_PKCS1_SHA512, &digest::SHA512,
                    &SHA512_PKCS1_DIGESTINFO_PREFIX,
                    "Signing using RSA with PKCS#1 1.5 padding and SHA-512.");
 
-macro_rules! rsa_pkcs1 {
+rsa_pss_padding!(RSA_PSS_SHA1, &digest::SHA1,
+                   "Signing using RSA with PSS padding and SHA-1.", 0);
+rsa_pss_padding!(RSA_PSS_SHA256, &digest::SHA256,
+                   "Signing using RSA with PSS padding and SHA-256.", 0);
+rsa_pss_padding!(RSA_PSS_SHA384, &digest::SHA384,
+                   "Signing using RSA with PSS padding and SHA-384.", 0);
+rsa_pss_padding!(RSA_PSS_SHA512, &digest::SHA512,
+                   "Signing using RSA with PSS padding and SHA-512.", 0);
+
+rsa_pss_padding!(RSA_PSS_SHA1_20, &digest::SHA1,
```

Comment:
Yes, the salt length should always be equal to the digest function output length, and the digest for the MGF should always be the same digest used for the hash. Basically, we are constraining ourselves to what TLS 1.3 and other recent IETF specs require: https://github.com/briansmith/ring/issues/115#issuecomment-227272106